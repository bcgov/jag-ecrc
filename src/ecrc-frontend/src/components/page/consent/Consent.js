import React, { useState, useEffect } from "react";
import PropTypes from "prop-types";

import Header from "../../base/header/Header";
import Footer from "../../base/footer/Footer";
import Declaration from "../../base/declaration/Declaration";
import { Button } from "../../base/button/Button";
import "../page.css";
import SideCard from "../../base/sideCard/SideCard";

export default function Consent({ page: { header }, onContinueClick }) {
  const [inputEnabled, setInputEnabled] = useState(
    "textinput_non_editable_gray"
  );
  const [firstBoxChecked, setFirstBoxChecked] = useState(false);
  const [secondBoxChecked, setSecondBoxChecked] = useState(false);
  const [thirdBoxChecked, setThirdBoxChecked] = useState(false);
  const [continueBtnEnabled, setContinueBtnEnabled] = useState(false);
  const [applicantName, setApplicantName] = useState("");

  useEffect(() => {
    if (firstBoxChecked && secondBoxChecked && thirdBoxChecked) {
      setInputEnabled("textinput_editable_white");
      if (applicantName) {
        setContinueBtnEnabled(true);
      } else {
        setContinueBtnEnabled(false);
      }
    } else {
      setInputEnabled("textinput_non_editable_gray");
      setContinueBtnEnabled(false);
    }
  }, [firstBoxChecked, secondBoxChecked, thirdBoxChecked, applicantName]);

  const backButton = {
    label: "Back",
    buttonStyle: "btn ecrc_accessary_btn",
    buttonSize: "btn",
    type: "submit"
  };

  const continueButton = {
    label: "Continue",
    buttonStyle: "btn ecrc_go_btn",
    buttonSize: "btn",
    type: "submit",
    disabled: !continueBtnEnabled
  };

  const contactSideCard = {
    heading: "Contact the Criminal Records Review Program",
    content: [],
    type: "contact",
    isWide: true
  };

  const noticeSideCard = {
    heading: "Collection Notice",
    content: [
      <div style={{ fontSize: "12px" }}>
        The Security Programs Division(SPD) will collect your personal
        information for the purpose of fulfilling the requirements of the
        Cannabis Control and Licensing Act(CCLA) and associated regulations in
        accordance with Sections 26(a) and(c) of the Freedom of Information and
        Protection of Privacy Act.Should you have any questions about the
        collection, use, or disclosure of personal information, please contact
        the Senior Policy Analyst, Security Programs Division via mail to PO Box
        9217 Stn Prov Govt Victoria, BC V8W 9J1; email to
        cannabissecurityscreening@gov.bc.ca; or by telephone at 1 - 855 - 587 -
        0185.
      </div>
    ],
    type: "notice",
    isWide: true
  };

  const textInput = {
    label: "Applicant Name",
    id: "textInputId",
    textInputStyle: inputEnabled
  };

  return (
    <main>
      <Header header={header} />
      <div className="page">
        <div className="content col-md-8">
          <h1>Consent for Criminal Record Check</h1>
          <Declaration
            style={{ paddingBottom: "30px" }}
            checkFirstBox={() => setFirstBoxChecked(!firstBoxChecked)}
            checkSecondBox={() => setSecondBoxChecked(!secondBoxChecked)}
            checkThirdBox={() => setThirdBoxChecked(!thirdBoxChecked)}
            textInput={textInput}
            onApplicantNameChange={setApplicantName}
          />
          <br />
          <div className="buttons" style={{ paddingLeft: "20px" }}>
            <Button
              button={backButton}
              onClick={() => {
                window.history.back();
              }}
            />
            <Button button={continueButton} onClick={onContinueClick} />
          </div>
        </div>

        <div className="sidecard">
          <SideCard key="contact" sideCard={contactSideCard} />
          <SideCard key="collectionnotice" sideCard={noticeSideCard} />
        </div>
      </div>
      <Footer />
    </main>
  );
}

Consent.propTypes = {
  page: PropTypes.shape({
    header: PropTypes.shape({
      name: PropTypes.string.isRequired
    }).isRequired
  }).isRequired,
  onContinueClick: PropTypes.func.isRequired
};
