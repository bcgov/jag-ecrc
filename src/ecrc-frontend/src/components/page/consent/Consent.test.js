import React from "react";
import { create } from "react-test-renderer";

import Consent from "./Consent";

describe("Consent Page Component", () => {
  test("Matches the snapshot", () => {
    const header = {
      name: "Criminal Record Check"
    };

    const page = {
      header
    };

    const onContinueClick = () => jest.fn();

    const consent = create(
      <Consent page={page} onContinueClick={onContinueClick} />
    );
    expect(consent.toJSON()).toMatchSnapshot();
  });
});
