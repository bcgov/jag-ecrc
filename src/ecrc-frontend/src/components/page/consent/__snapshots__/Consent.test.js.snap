// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Consent Page Component Matches the snapshot 1`] = `
<main>
  <header>
    <nav
      className="navbar navbar-expand-lg navbar-dark"
    >
      <div
        className="container"
      >
        <a
          className="navbar-brand"
          href="/"
        >
          <img
            alt="B.C. Government Logo"
            className="img-fluid d-none d-md-block"
            height="44"
            src="/images/bcid-logo-rev-en.svg"
            width="181"
          />
          <img
            alt="B.C. Government Logo"
            className="img-fluid d-md-none"
            height="44"
            src="/images/bcid-symbol-rev.svg"
            width="64"
          />
        </a>
        <div>
          <a
            className="navbar-brand nav-item nav-link"
            href="/"
          >
            Criminal Record Check
          </a>
        </div>
        <button
          aria-controls="navbarNavAltMarkup"
          aria-expanded="false"
          aria-label="Toggle navigation"
          className="navbar-toggler"
          data-target="#navbarNavAltMarkup"
          data-toggle="collapse"
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          className="collapse navbar-collapse"
          id="navbarNavAltMarkup"
        />
      </div>
    </nav>
  </header>
  <div
    className="page"
  >
    <div
      className="content col-md-8"
    >
      <h1>
        Consent for Criminal Record Check
      </h1>
      <p>
        In this section, you consent to a criminal background check. Please read the declaration before agreeing.
      </p>
      <p>
        You must complete all mandatory fields (
        <span
          className="mandatory"
          id="asterisk"
        >
          *
        </span>
         ):
      </p>
      <div>
        <div
          className="declareTitle"
        >
          CONSENT
        </div>
        <section
          className="declareSection"
        >
          <input
            onClick={[Function]}
            type="checkbox"
          />
          <span
            className="declaration-cb"
          >
            I, the undersigned, do hereby consent to the collection and disclosure by the Royal Canadian Mounted Police (RCMP) and other law enforcement agencies, as well as other duly authorized agencies of the government, of any and all information related to the security screening checks in support of this application for up to two years. This may include some or all of: (a) criminal record check or fingerprint-based criminal record verification by searching the Canadian Police Information Centre database; (b) a police information check; (c) a check of intelligence databases maintained by law enforcement agencies; (d) a check of records in the justice information system of the Ministry of Attorney General and; (e) a check of records in the corrections information system of the Ministry of Public Safety and Solicitor General.
          </span>
        </section>
        <section
          className="declareSection"
        >
          <input
            onClick={[Function]}
            type="checkbox"
          />
          <span
            className="declaration-cb"
          >
            I certify that, to the best of my knowledge, the information I have provided on my application and will provide as necessary is complete, honest and accurate. I understand that a false statement or omission of facts herein may lead to a denial of a cannabis workers registration. I am also aware that later discovery of an omission or misrepresentation may be grounds for any finding of suitability to be suspended or revoked.
          </span>
        </section>
        <section
          className="declareSection"
        >
          <input
            onClick={[Function]}
            type="checkbox"
          />
          <span
            className="declaration-cb"
          >
            The third checkbox for my application and will provide as necessary is complete.
          </span>
        </section>
      </div>
      <br />
      <div
        className="buttons"
        style={
          Object {
            "paddingLeft": "20px",
          }
        }
      >
        <button
          className="btn ecrc_accessary_btn btn"
          onClick={[Function]}
          type="submit"
        >
          Cancel and Exit
          <div
            className="loader-hide"
          >
            <div
              className="btn-loader"
            />
          </div>
        </button>
        <button
          className="btn ecrc_go_btn btn"
          disabled={true}
          onClick={[Function]}
          type="submit"
        >
          Continue
          <div
            className="loader-hide"
          >
            <div
              className="btn-loader"
            />
          </div>
        </button>
      </div>
    </div>
    <div
      className="sidecard"
    >
      <div>
        <div
          className="wide-dashboard-spacing"
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <div
            className="row"
          >
            <div
              className="col-12"
              style={
                Object {
                  "paddingTop": "30px",
                  "position": "relative",
                }
              }
            >
              <p />
              <section
                className="submit-container"
                id="blue-section"
              >
                <h2
                  className="heading-style"
                >
                  <div
                    className="row"
                  >
                    <div
                      className="round-icon-wrapper"
                    >
                      <svg
                        className="side-card-icon"
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        strokeWidth="0"
                        style={
                          Object {
                            "color": undefined,
                          }
                        }
                        viewBox="0 0 512 512"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
                        />
                      </svg>
                    </div>
                    <div
                      className="sideCardTitle"
                    >
                      Contact the Criminal Records Review Program
                    </div>
                  </div>
                </h2>
                <div
                  className="submit-content"
                >
                  <span>
                    <div
                      className="side-card-text"
                    >
                      For questions about criminal record checks, contact the Criminal Records Review Program Monday to Friday, 8:30 a.m. - 4:30 p.m.
                    </div>
                    <div
                      className="side-card-text"
                    >
                      <br />
                      <div>
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          strokeWidth="0"
                          style={
                            Object {
                              "color": undefined,
                            }
                          }
                          viewBox="0 0 512 512"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M64 128H32c-17.67 0-32 14.33-32 32v320c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V160c0-17.67-14.33-32-32-32zm416 32V77.25c0-8.49-3.37-16.62-9.37-22.63L425.37 9.37c-6-6-14.14-9.37-22.63-9.37H160c-17.67 0-32 14.33-32 32v448c0 17.67 14.33 32 32 32h320c17.67 0 32-14.33 32-32V192c0-17.67-14.33-32-32-32zM288 432c0 8.84-7.16 16-16 16h-32c-8.84 0-16-7.16-16-16v-32c0-8.84 7.16-16 16-16h32c8.84 0 16 7.16 16 16v32zm0-128c0 8.84-7.16 16-16 16h-32c-8.84 0-16-7.16-16-16v-32c0-8.84 7.16-16 16-16h32c8.84 0 16 7.16 16 16v32zm128 128c0 8.84-7.16 16-16 16h-32c-8.84 0-16-7.16-16-16v-32c0-8.84 7.16-16 16-16h32c8.84 0 16 7.16 16 16v32zm0-128c0 8.84-7.16 16-16 16h-32c-8.84 0-16-7.16-16-16v-32c0-8.84 7.16-16 16-16h32c8.84 0 16 7.16 16 16v32zm16-112H176V48h208v32c0 8.84 7.16 16 16 16h32v96z"
                          />
                        </svg>
                        Â 
                        <div
                          className="contact-title"
                        >
                          Fax: 
                        </div>
                        (250)356-1889
                      </div>
                      <div>
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          strokeWidth="0"
                          style={
                            Object {
                              "color": undefined,
                            }
                          }
                          viewBox="0 0 512 512"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M493.4 24.6l-104-24c-11.3-2.6-22.9 3.3-27.5 13.9l-48 112c-4.2 9.8-1.4 21.3 6.9 28l60.6 49.6c-36 76.7-98.9 140.5-177.2 177.2l-49.6-60.6c-6.8-8.3-18.2-11.1-28-6.9l-112 48C3.9 366.5-2 378.1.6 389.4l24 104C27.1 504.2 36.7 512 48 512c256.1 0 464-207.5 464-464 0-11.2-7.7-20.9-18.6-23.4z"
                          />
                        </svg>
                        Â 
                        <div
                          className="contact-title"
                        >
                          Office: 
                        </div>
                        Toll free - 1 855 587-0185 (press option 2)
                      </div>
                      <div>
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          strokeWidth="0"
                          style={
                            Object {
                              "color": undefined,
                            }
                          }
                          viewBox="0 0 512 512"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
                          />
                        </svg>
                         Â 
                        <div
                          className="contact-title"
                        >
                          Email: 
                        </div>
                        <a
                          className="email"
                          href="mailto:criminalrecords@gov.bc.ca"
                        >
                          criminalrecords@gov.bc.ca
                        </a>
                      </div>
                    </div>
                  </span>
                </div>
              </section>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          className="wide-dashboard-spacing"
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <div
            className="row"
          >
            <div
              className="col-12"
              style={
                Object {
                  "paddingTop": "30px",
                  "position": "relative",
                }
              }
            >
              <p />
              <section
                className="submit-container"
                id="notice-section"
                style={
                  Object {
                    "backgroundColor": "#F2F2F2",
                    "border": "none",
                    "color": "#000",
                  }
                }
              >
                <h2
                  style={
                    Object {
                      "color": "#000",
                    }
                  }
                >
                  Collection Notice
                </h2>
                <div
                  className="submit-content"
                >
                  <span>
                    <div
                      style={
                        Object {
                          "fontSize": "12px",
                        }
                      }
                    >
                      The Security Programs Division will collect your personal information for the purpose of fulfilling the criminal record check requirements of the Criminal Records Review Act (CRRA) and in accordance with sections 26(a) and (c) of the Freedom of Information and Protection of Privacy Act (FoIPPA) and section 6 of the CRRA . Additionally, SPD may collect personal information under section 26(e) of FoIPPA for the purpose of evaluating the Criminal Records Review Program and activities to better serve you. Should you have any questions about the collection, use, or disclosure of personal information, please contact the Criminal Records Review Program, Security Programs Division via mail to PO Box 9217 Stn Prov Govt Victoria, BC V8W 9J1; email toÂ 
                      <a
                        href="mailto:criminalrecords@gov.bc.ca"
                      >
                        criminalrecords@gov.bc.ca
                      </a>
                      Â or by telephone at 1- 855-587-0185 (option 2).
                    </div>
                  </span>
                </div>
              </section>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <footer
    className="footer"
  >
    <nav
      className="navbar navbar-expand-sm navbar-dark"
    >
      <div
        className="container"
      >
        <ul
          className="navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/disclaimer"
              rel="noopener noreferrer"
              target="_blank"
            >
              Disclaimer
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/privacy"
              rel="noopener noreferrer"
              target="_blank"
            >
              Privacy
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/accessibility"
              rel="noopener noreferrer"
              target="_blank"
            >
              Accessibility
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/copyright"
              rel="noopener noreferrer"
              target="_blank"
            >
              Copyright
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/contact-us"
              rel="noopener noreferrer"
              target="_blank"
            >
              Contact Us
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </footer>
</main>
`;
